Netable/Netable/Netable.swift

```diff
@@ -125,7 +125,15 @@ open class Netable {
                 switch decoded {
                 case .success(let raw):
                     let finalizedData = request.finalize(raw: raw)
-                    self.log(.requestSuccess(request: requestInfo, taskTime: time, statusCode: response.statusCode, responseData: data, finalizedResult: finalizedData))
+
+                    if case let .success(finalizedResult) = finalizedData {
+                        self.log(.requestSuccess(request: requestInfo, taskTime: time, statusCode: response.statusCode, responseData: data, finalizedResult: finalizedResult))
+                    } else if case let .failure(finalizedError) = finalizedData {
+                        throw NetableError.decodingError(finalizedError, data)
+                    } else {
+                        throw NetableError.resourceExtractionError("Failed to parse the result of `Finalize`.")
```

Comment:
Can it ever actually fall though into this case. It seems like success and failure should be exhaustive and I thought that those parameters were non optional?

Maybe the issue is if the user makes them optional, like sets the FinalResource type to Int? (or whatever), HOWEVER, it seems like in that case we DON'T want to treat a nil as a failure. if the user makes the FinalResource type `Int?` and then passes back `.success(nil)` we probably should NOT treat that as a resource extraction failure, which I think this code might be doing.

Reply:
Yeah you're right, we can do this a lot more simply with a switch. Updated.